### PR TITLE
Update Auth templates for CurrentUser types and JWT/Error imports

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes https://github.com/redwoodjs/redwood/issues/3596

This PR complete an audit of the auth templates to remove TS errors and warnings for the currentUser and roles.

* When needed, a RedwoodUser type defines a flexible set of attributes and the optional list of roles. This is for Supabase and Netlify mainly.
* For DbAuth, since  the User model doesn't ye have roles, the role check is commented out and the developer is informed to re-enabled them as needed
* In other templates, various imports were corrected to import JWT parsing and errors from the proper packages

### CODE REVIEW

**Comment by thedavidprice** [comment]
@dthyresson merge conflict due to recent changes to dbAuth template

@dac09 ping for your review to include in v0.39

File: packages/cli/src/commands/setup/auth/templates/auth.ts.template
Comment (dac09) [inline packages/cli/src/commands/setup/auth/templates/auth.ts.template:9]: I would use unknown here,instead of any

**Comment by dac09** [comment]
@dthyresson looks ok to me, apart from that one minor comment. I also wasn't sure how to test it? I'm not sure when `
packages/cli/src/commands/setup/auth/templates/auth.ts.template` is used 🤷

File: packages/cli/src/commands/setup/auth/templates/auth.ts.template
Comment (dthyresson) [reply packages/cli/src/commands/setup/auth/templates/auth.ts.template:9]: Will do. That means I can also remove the lint disable.

**Comment by dthyresson** [comment]
> @dthyresson looks ok to me, apart from that one minor comment. I also wasn't sure how to test it? I'm not sure when ` packages/cli/src/commands/setup/auth/templates/auth.ts.template` is used 🤷

I test by creating a new RW project and then adding a `lib/auth.ts` and then copying over this template content -- and check for TypeScript issues.

And currently there are none:

image

**Comment by thedavidprice** [comment]
This looks good and I need it in for PR #2896

Will merge after the Babel PR is in (waiting for CI) unless you gents say otherwise.

**Review by thedavidprice — APPROVED** [review]

**Pull request merged**
